hpx/runtime/components/migrate_component.hpp

```diff
@@ -48,8 +49,8 @@ namespace hpx { namespace components
     migrate(naming::id_type const& to_migrate,
         naming::id_type const& target_locality)
     {
-        typedef server::trigger_migrate_component_action<Component> action_type;
-        return async<action_type>(naming::get_locality_from_id(to_migrate),
+        typedef server::perform_migrate_component_action<Component> action_type;
+        return hpx::detail::async_colocated<action_type>(to_migrate,
```

Comment:
It's an internal invocation, so I directly call the implementation for the colocated async.